daemon/run.go

```diff
@@ -0,0 +1,319 @@
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+
+	"github.com/akitasoftware/akita-cli/apispec"
+	"github.com/akitasoftware/akita-cli/har_loader"
+	"github.com/akitasoftware/akita-cli/plugin"
+	"github.com/akitasoftware/akita-cli/rest"
+	"github.com/akitasoftware/akita-cli/trace"
+	"github.com/akitasoftware/akita-cli/util"
+	"github.com/akitasoftware/akita-libs/akid"
+	kgxapi "github.com/akitasoftware/akita-libs/api_schema"
+	"github.com/akitasoftware/akita-libs/sampled_err"
+)
+
+const WITNESS_BUFFER_SIZE = 10_000
+
+type Witness = har_loader.CustomHAREntry
+
+type Args struct {
+	// Required args.
+	ClientID akid.ClientID
+	Domain   string
+
+	// Optional args.
+	PortNumber uint16
+
+	Plugins []plugin.AkitaPlugin
+}
+
+// A wrapper around HTTP request handlers for handling HTTP responses generically.
+type httpHandler func(*http.Request) HTTPResponse
```

Comment:
I don't understand why the return type is HTTPResponse rather than *HTTPResponse (maybe it's to do with my confusion below about how it all works?)  It seems to require a lot of un-idiomatic pointer dereferencing, i.e, `*NewHTTPResponse(...)` instead of just `NewHttpResponse(...)`.

Reply:
The return type is `HTTPResponse` because I want to force implementations to provide an actual response; they shouldn't be able to return `nil`. In 638cf0a, I've tweaked things so we have fewer pointer dereferences now.